addons/mass_mailing_sms/tests/test_mailing_retry.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.mass_mailing_sms.tests.common import MassSMSCommon
+from odoo.addons.base.tests.test_ir_cron import CronMixinCase
+from odoo.tests.common import users
+
+from unittest.mock import patch
+
+class TestMailingRetrySMS(MassSMSCommon, CronMixinCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestMailingRetrySMS, cls).setUpClass()
+        cls._create_mailing_list()
+
+    @users('user_marketing')
+    def test_sms_retry_immediate_trigger(self):
+        mailing = self.env['mailing.mailing'].create({
+            'name': 'TestMailing',
+            'subject': 'Test',
+            'mailing_type': 'sms',
+            'body_plaintext': 'Coucou hibou',
+            'mailing_model_id': self.env['ir.model']._get('res.partner').id,
+            'contact_list_ids': [(4, self.mailing_list_1.id)],
+        })
+
+        # force the SMS sending to fail to test our retry mechanism
+        def patched_mail_mail_send(mail_records, auto_commit=False, raise_exception=False):
+            mail_records.write({'state': 'exception', 'failure_reason': 'forced_failure'})
+
+        with patch('odoo.addons.mail.models.mail_mail', patched_mail_mail_send):
```

Comment:
When I said something similar, I meant by example by patching the "_send" method in the sms.sms model.

Here you are patching mail.mail, which is not relevant when sending sms'es.